WindowsAzureMessaging/WindowsAzureMessaging/Helpers/SBNotificationHubHelper.m

```diff
@@ -0,0 +1,247 @@
+//----------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//----------------------------------------------------------------
+
+#import "SBNotificationHubHelper.h"
+#import "SBTemplateRegistration.h"
+#import <CommonCrypto/CommonDigest.h>
+#import <CommonCrypto/CommonHMAC.h>
+
+@implementation SBNotificationHubHelper
+
+static const char encodingTable[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+static char decodingTable[128];
+static NSString *decodingTableLock = @"decodingTableLock";
+static NSString *const domain = @"WindowsAzureMessaging";
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
++ (NSString *)urlEncode:(NSString *)urlString {
+    return (__bridge NSString *)CFURLCreateStringByAddingPercentEscapes(kCFAllocatorDefault, (__bridge CFStringRef)urlString, NULL,
```

Comment:
It seems that it should be `__bridge_transfer` to transfer ownership and avoid leak

Reply:
Fixed